subsys/net/ip/route.c

```diff
@@ -713,9 +766,13 @@ net_route_mcast_lookup(struct in6_addr *group)
 		struct net_route_entry_mcast *route = &route_mcast_entries[i];
 
 		if (!route->is_used) {
-			if (net_ipv6_addr_cmp(group, &route->group)) {
-				return route;
-			}
+			continue;
+		}
+
+		if (net_ipv6_is_prefix(group->s6_addr,
+							route->group.s6_addr,
```

Comment:
There is indentation issue here. Do you have tab size set to 8 spaces, that might explain the indentation problems.